examples/use_as_library/instance_segmentation.py

```diff
@@ -25,17 +36,21 @@ def cell_segmentation():
     # we enable changing the hyperparameters, e.g. 'pred_iou_thresh', without recomputing masks and embeddings,
     # to support (interactive) evaluation of different hyperparameters.
 
-    # Create the automatic mask generator class.
-    amg = instance_segmentation.AutomaticMaskGenerator(predictor)
-
-    # Initialize the mask generator with the image and the pre-computed embeddings.
-    amg.initialize(image, embeddings, verbose=True)
+    # Step 1: Get the 'predictor' and 'segmenter' to perform automatic instance segmentation.
+    predictor, segmenter = get_predictor_and_segmenter(
+        model_type=model_type,  # choice of the Segment Anything model
+        checkpoint=None,  # overwrite to pass your own finetuned model.
+        device=None,  # overwrite to pass device to run the model inference. by default, chooses best supported device.
+        amg=True,  # set the automatic segmentation mode to AMG.
```

Comment:
Let's not hard-code `amg=True` here, but rather use the default so that it uses AIS if we have a model with decoder.